hedera-node/src/main/java/com/hedera/services/fees/calculation/consensus/txns/UpdateTopicResourceUsage.java

```diff
@@ -46,33 +46,36 @@ public boolean applicableTo(TransactionBody txn) {
     }
 
     @Override
-    public FeeData usageGiven(TransactionBody txn, SigValueObj sigUsage, StateView view) throws InvalidTxBodyException {
-        try {
-            MerkleTopic merkleTopic = view.topics().get(MerkleEntityId.fromTopicId(txn.getConsensusUpdateTopic().getTopicID()));
-            long rbsIncrease = getUpdateTopicRbsIncrease(
-                    txn.getTransactionID().getTransactionValidStart(),
-                    JKey.mapJKey(merkleTopic.getAdminKey()),
-                    JKey.mapJKey(merkleTopic.getSubmitKey()),
-                    merkleTopic.getMemo(),
-                    merkleTopic.hasAutoRenewAccountId(),
-                    lookupExpiry(merkleTopic),
-                    txn.getConsensusUpdateTopic());
-            return getConsensusUpdateTopicFee(txn, rbsIncrease, sigUsage);
-        } catch (Exception illegal) {
-            log.warn("Usage estimation unexpectedly failed for {}!", txn, illegal);
-            throw new InvalidTxBodyException(illegal);
+    public FeeData usageGiven(TransactionBody txnBody, SigValueObj sigUsage, StateView view) throws InvalidTxBodyException {
+        if(txnBody == null) {
+            throw new InvalidTxBodyException("consensusCreateTopic field not available for Fee Calculation");
+        } else {
+            try {
+                MerkleTopic merkleTopic = view.topics().get(
+                        MerkleEntityId.fromTopicId(txnBody.getConsensusUpdateTopic().getTopicID()));
+                if (merkleTopic != null && merkleTopic.hasAdminKey()) {
+                    long rbsIncrease = getUpdateTopicRbsIncrease(
+                            txnBody.getTransactionID().getTransactionValidStart(),
+                            JKey.mapJKey(merkleTopic.getAdminKey()),
+                            JKey.mapJKey(merkleTopic.getSubmitKey()),
+                            merkleTopic.getMemo(),
+                            merkleTopic.hasAutoRenewAccountId(),
+                            lookupExpiry(merkleTopic),
+                            txnBody.getConsensusUpdateTopic());
+                    return getConsensusUpdateTopicFee(txnBody, rbsIncrease, sigUsage);
```

Comment:
I think only the `getUpdateTopicRbsIncrease` function needs try-catch. Moving everything else out.